ResearchKit/Charts/ORKChartTypes.h

```diff
@@ -0,0 +1,150 @@
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import <UIKit/UIKit.h>
+#import "ORKDefines.h"
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol ORKGraphChartType
+
+/**
+ A Boolean value indicating that the value is unset. (read-only)
+ */
+@property (nonatomic, readonly) BOOL isUnset;
+
+@end
+
+
+/**
+ The `ORKValueRange` class represents a range between two values of the double type. It can be used
+ in graph chart plots to draw value ranges.
+ */
+ORK_CLASS_AVAILABLE
+@interface ORKValueRange : NSObject <ORKGraphChartType>
+
+/**
+ Returns a value range initialized using the specified `minimumValue` and `maximumValue`.
+ 
+ @param minimumValue     The `minimumValue` to set.
+ @param maximumValue     The `maximumValue` to set.
+ 
+ @return A value range object.
+ */
+- (instancetype)initWithMinimumValue:(double)minimumValue maximumValue:(double)maximumValue NS_DESIGNATED_INITIALIZER;
+
+/**
+ Returns a value range initialized using the specified `value` for both `minimumValue` and
+ `maximumValue`. This is useful for plotting data points that model a single value with no range.
+ 
+ This method is a convenience initializer.
+ 
+ @param value    The `minimumValue` and `maximumValue` to set.
+ 
+ @return A value range object.
+ */
+- (instancetype)initWithValue:(double)value;
+
+/**
+ Returns a value range initialized using the `ORKDoubleInvalidValue` value for both `minimumValue`
+ and `maximumValue`. This denotes an unset or invalid value range. It is useful, for example, for
+ representing unavailable data in discontinous graph chart plots.
+ 
+ This method is a convenience initializer.
+ 
+ @return A value range object initialized with the `ORKDoubleInvalidValue` value.
+ */
+- (instancetype)init;
+
+/**
+ The upper limit of the value range.
+ */
+@property (nonatomic) double maximumValue;
+
+/**
+ The lower limit of the value range.
+ */
+@property (nonatomic) double minimumValue;
+
+/**
+ A Boolean value indicating that `minimumValue` is equal to `maximumValue`. (read-only)
+ */
+@property (nonatomic, readonly) BOOL isEmptyRange;
+
+/**
+ A Boolean value indicating that both `minimum value` and `maximum value` are equal to the
+ `ORKDoubleInvalidValue` value. (read-only)
+ */
+@property (nonatomic, readonly) BOOL isUnset;
+
+@end
+
+
+/**
+ The `ORKValueStack` class represents an arbitrary list of values of the double type. It can be used
+ in graph chart plots to draw stacked values.
+ */
+ORK_CLASS_AVAILABLE
+@interface ORKValueStack : NSObject <ORKGraphChartType>
+
+/**
+ Returns an empty value stack.
+ 
+ @return An empty value stack object.
+ */
+- (instancetype)init;
+
+/**
+ Returns an value stack initialized with the passed double values.
+ 
+ @param value   A `nil` terminated list of `double` values.
+ 
+ @return A value stack object initialized with the passed double values.
+ */
+- (instancetype)initWithStackedValueArray:(NSArray<NSNumber *> *)stackedValues NS_DESIGNATED_INITIALIZER;
```

Comment:
This is related to the discussion above of renaming `ORKValueRange` -> `ORKRangeValue` and `ORKValueStack` -> `ORKStackValue`, and problematic for the same reason I state above.